RiotSharp/RiotApi.cs

```diff
@@ -167,17 +167,18 @@ public List<Summoner> GetSummoners(Region region, List<long> summonerIds)
         /// <returns>A list of summoners.</returns>
         public async Task<List<Summoner>> GetSummonersAsync(Region region, List<long> summonerIds)
         {
-            var list = new List<Summoner>();
-            foreach (var grp in MakeGroups(summonerIds, MaxNrSummoners))
-            {
-                var json = await requester.CreateGetRequestAsync(
-                    string.Format(SummonerRootUrl,
-                        region.ToString()) + string.Format(IdUrl, Util.BuildIdsString(grp)),
-                    region);
-                var subList = (await Task.Factory.StartNew(() =>
-                    JsonConvert.DeserializeObject<Dictionary<long, Summoner>>(json))).Values.ToList();
-                list.AddRange(subList);
-            }
+            var tasks = MakeGroups(summonerIds, MaxNrSummoners).Select(
+                grp => requester.CreateGetRequestAsync(
+                    string.Format(SummonerRootUrl, region.ToString()) +
+                    string.Format(IdUrl, Util.BuildIdsString(grp)), region
+                    ).ContinueWith(
+                        json => JsonConvert.DeserializeObject<Dictionary<long, Summoner>>(json.Result)
+                    )
+                ).ToList();
+
+            await Task.WhenAll(tasks);
```

Comment:
The rest of the requests will still be sent, but after all requests complete the `SelectMany(task => task.Result)` would cause the error to be thrown, IIRC